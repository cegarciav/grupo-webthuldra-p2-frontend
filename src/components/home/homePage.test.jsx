import React from 'react';
import PropTypes from 'prop-types';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MainRouter from '../../routers/index';

function TestRouter({ path }) {
  return (
    <MemoryRouter initialEntries={[path]}>
      <MainRouter />
    </MemoryRouter>
  );
}

TestRouter.propTypes = {
  path: PropTypes.string.isRequired,
};

describe('HomePage', () => {
  describe('When user is not logged-in', () => {
    it('renders the home page', () => {
      const tree = renderer.create(<TestRouter path="/" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
