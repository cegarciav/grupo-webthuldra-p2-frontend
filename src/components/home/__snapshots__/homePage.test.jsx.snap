// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage When user is not logged-in renders the home page 1`] = `
<main
  id="home-page"
>
  <a
    className="main-link"
    href="/stores"
    onClick={[Function]}
  >
    Explorar tiendas
  </a>
  <a
    className="main-link"
    href="/signup"
    onClick={[Function]}
  >
    Registrarse
  </a>
</main>
`;
